[Ye Xinjian] iP

XXJJXJ opened the pull request:
# CS2103T iP

Application name: ~~Duke~~ **FullOfBugs**
Author: Ye Xinjian

#### Description
FullOfBugs is a simple CLI app that helps you remember your tasks!

All you need to do is,

1. Download it from [here](https://github.com/XXJJXJ/ip/releases).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

## Current Features:
- Add tasks to list
- Mark task as done
- List out all tasks
- Delete task
- Find task(s)

## Coming Soon:
- GUI

## Todo list:
- [x] Add Gradle Support
- [x] Add CheckStyle
- [x] GUI

## Others
> "Programming is all about copy and pasting":+1: - Me, Myself and I

If you are interested in my `Duke` program here is a sneak peek of my `Duke`'s constructor:

```java
public Duke(String filePath) {
    ui = new Ui();
    this.store = new Storage(filePath);
    try {
        taskList = new TaskList(store.load());
    } catch (DukeFileException e) {
        ui.showError(e.getMessage());
        taskList = new TaskList();
    }
}
```

Commit:
Add Gradle support

XXJJXJ committed:
Complete Level 1. Greet, Echo, Exit

XXJJXJ committed:
Complete Level 2. Add, List

XXJJXJ committed:
Complete Level 3. Mark as Done

XXJJXJ committed:
Complete Level 3. Mark as Done

XXJJXJ committed:
Complete Level 4. ToDos, Events, Deadlines

XXJJXJ committed:
Complete A-TextUiTesting: Automated Text UI Testing

XXJJXJ committed:
Complete Level 5. Handle Errors

XXJJXJ committed:
Improve Level 5. Handle Errors

XXJJXJ committed:
Complete Level 6. Delete

XXJJXJ committed:
Improve exception detection

XXJJXJ committed:
Improve exception handling

XXJJXJ committed:
Complete Level 7. Save

XXJJXJ committed:
Fix bugs for Level 7

XXJJXJ committed:
Improve code readability

XXJJXJ committed:
Implement Directory and File Creation methods

XXJJXJ committed:
Remove generated directory and txt file

XXJJXJ committed:
Complete Level 8. Dates and Times

XXJJXJ committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

XXJJXJ committed:
Resolve more merging bugs from Level 8

XXJJXJ committed:
Update .gitignore

XXJJXJ committed:
Fix Expected output

XXJJXJ committed:
Create TaskList and Ui

XXJJXJ committed:
Add more classes

XXJJXJ committed:
Minor adjustment to messages

XXJJXJ committed:
Add class header docs

XXJJXJ committed:
Revamp Duke logic

XXJJXJ committed:
Fix TaskList bug

XXJJXJ committed:
Improve exception detection

XXJJXJ committed:
Seperate files into respective packages

XXJJXJ committed:
Add comparison methods for Command and Tasks

XXJJXJ committed:
Add toString() methods to Command classes

XXJJXJ committed:
Update .gitignore to ignore test storage txt file

XXJJXJ committed:
Add JUnit test for Parser

XXJJXJ committed:
Add JUnit test for Command

XXJJXJ committed:
Add JUnit for Task

XXJJXJ committed:
Update runtest.bat after packaging

XXJJXJ committed:
Update EXPECTED.txt

XXJJXJ committed:
ParserTest: Update imports to remove .*

XXJJXJ committed:
Ui: Update messages to use 4 spaces instead of 1 tab

XXJJXJ committed:
Storage: Update variable names

XXJJXJ committed:
TaskList: Rectify indentations

XXJJXJ committed:
Parser: Fix indentation

XXJJXJ committed:
Storage: Update comments

XXJJXJ committed:
TaskNotFoundException: Replace tab with 4 spaces

XXJJXJ committed:
ParseTest: Add new test cases

XXJJXJ committed:
Add JUnit test for DukeExceptions

XXJJXJ committed:
Add META-INF directory to .gitignore

XXJJXJ committed:
Add Javadocs to classes in duke package

XXJJXJ committed:
Refactor classes in duke to adhere to coding standards

XXJJXJ committed:
Refactor classes in duke.commands to adhere to coding standards

XXJJXJ committed:
Refactor classes in duke.exceptions to adhere to coding standards

XXJJXJ committed:
Refactor classes in duke.task to adhere to coding standards

XXJJXJ committed:
Refactor TaskTest to adhere to coding standards

XXJJXJ committed:
Add Javadocs for classes in duke.commands

XXJJXJ committed:
Add Javadocs to classes in duke.exceptions

XXJJXJ committed:
Add Javadocs to Task

XXJJXJ committed:
Refactor Task into an abstract class

XXJJXJ committed:
Add Javadocs to classes in duke.task

XXJJXJ committed:
Add FindCommand class

XXJJXJ committed:
Refactor classes to include FindCommand

XXJJXJ committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/commands/Command.java
#	src/main/java/duke/exceptions/CommandParamException.java
#	src/main/java/duke/exceptions/EmptyDescriptionException.java
#	src/main/java/duke/exceptions/TaskIsCompleteException.java
#	src/main/java/duke/exceptions/TaskNotFoundException.java

XXJJXJ committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/exceptions/TaskNotFoundException.java

XXJJXJ committed:
Remove redundant method showLoadingError from Ui

XXJJXJ committed:
Parser: Change variable names to be more meaningful

XXJJXJ committed:
Update DukeExceptionTest class

XXJJXJ committed:
AddCommand: Change variable name t to task

XXJJXJ committed:
Storage: Update comments

XXJJXJ committed:
EmptyDescriptionException: Remove line spacing

XXJJXJ committed:
Merge branch 'master' into add-gradle-support

XXJJXJ committed:
Merge branch 'add-gradle-support'

XXJJXJ committed:
Add Checkstyle configuration files

XXJJXJ committed:
Update build.gradle mainClassName

XXJJXJ committed:
Duke.java: Fix style violations

XXJJXJ committed:
Parser.java: Fix import order violation

XXJJXJ committed:
Storage.java: Fix import order violation

XXJJXJ committed:
TaskList.java: Fix import order violation

XXJJXJ committed:
Ui.java: Fix import order violation

XXJJXJ committed:
AddCommand.java: Fix import order violation

XXJJXJ committed:
Command.java: Fix import order violation

XXJJXJ committed:
DeleteCommand.java: Fix import order violation

XXJJXJ committed:
DoneCommand.java: Fix import order violation

XXJJXJ committed:
ExitCommand.java: Fix import order violation

XXJJXJ committed:
FindCommand.java: Fix import order violation

XXJJXJ committed:
ListCommand.java: Fix import order violation

XXJJXJ committed:
Deadline.java: Fix import order violation

XXJJXJ committed:
Event.java: Fix import order violation

XXJJXJ committed:
ParserTest.java: Fix import order violation

XXJJXJ committed:
CommandTest.java: Fix import order violation

XXJJXJ committed:
DukeExceptionTest.java: Fix import order violation

XXJJXJ committed:
TaskTest.java: Fix style violations

XXJJXJ committed:
Merge branch 'A-CheckStyle'

XXJJXJ committed:
Configure build.gradle to include javaFx

XXJJXJ committed:
Configure build.gradle to include javaFx

XXJJXJ committed:
Add avatar pictures

XXJJXJ committed:
Add avatar pictures

XXJJXJ committed:
Move Parser.java to util package

XXJJXJ committed:
Move Parser.java to util package

XXJJXJ committed:
Move Storage.java into util package

XXJJXJ committed:
Move Storage.java into util package

XXJJXJ committed:
Move Ui.java into util package

XXJJXJ committed:
Move Ui.java into util package

XXJJXJ committed:
Move TaskList.java into util package

XXJJXJ committed:
Move TaskList.java into util package

XXJJXJ committed:
Update imports to util package

XXJJXJ committed:
Update imports to util package

XXJJXJ committed:
Modify exception messages

XXJJXJ committed:
Modify exception messages

XXJJXJ committed:
Add DialogBox class

XXJJXJ committed:
Add DialogBox class

XXJJXJ committed:
Add Launcher class

XXJJXJ committed:
Add Launcher class

XXJJXJ committed:
Add MainWindow class

XXJJXJ committed:
Add MainWindow class

XXJJXJ committed:
Add Main class

XXJJXJ committed:
Add Main class

XXJJXJ committed:
Add FXML files for Gui

XXJJXJ committed:
Add FXML files for Gui

XXJJXJ committed:
Implement Gui

XXJJXJ committed:
Implement Gui

XXJJXJ committed:
Update README.md

XXJJXJ committed:
Update README.md

XXJJXJ committed:
Ui.java: Remove redundant methods

XXJJXJ committed:
Ui.java: Remove redundant methods

XXJJXJ committed:
Duke.java: Change class fields to final

XXJJXJ committed:
Duke.java: Change class fields to final

benluiwj commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+        ui.showWelcomeMessage();
+        boolean isExit = false;
+
+        while(!isExit) {
```
Should there be a whitespace following 'while' here?

XXJJXJ replied:
Yes, thank you for pointing out my errors! Fixed it!

benluiwj commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+
+    public static void main(String[] args) {
+        java.nio.file.Path filepath = java.nio.file.Paths.get("src",
+                "main", "java","data","StoredData.txt");
```
Should there be a whitespace after "java" and before ','  as well as for "data"? ie`"java", "data", ...`

benluiwj commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.UnknownCommandException;
+
+import java.util.Scanner;
```
Should this import statement be here? ie perhaps it could be something like this 
```
package duke;

import java.util.Scanner;

//other imports
```

benluiwj commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.TaskList;
+import duke.Storage;
```
Is the lexicographical order wrong here? Perhaps it should be something like this : 
```
//other imports
duke.execeptions.UnknownCommandException;
duke.Storage;
duke.task.task.ToDo;
```

benluiwj reviewed (commented):
Looks good 👍 
Some importing statement and minor coding standards changes requested

XXJJXJ committed:
Remove META-INF from .gitignore

XXJJXJ committed:
Remove META-INF from .gitignore

XXJJXJ committed:
Add MANIFEST.MF

XXJJXJ committed:
Add MANIFEST.MF

XXJJXJ committed:
Duke.java: Update and remove redundant methods

XXJJXJ committed:
Duke.java: Update and remove redundant methods

XXJJXJ committed:
Add close Gui feature with exit command

XXJJXJ committed:
Add close Gui feature with exit command

XXJJXJ committed:
Main.java: Delete comment

XXJJXJ committed:
Main.java: Delete comment

XXJJXJ committed:
Fix coding standard violations

XXJJXJ committed:
Fix coding standard violations

XXJJXJ committed:
Merge branch 'branch-Level-10'

XXJJXJ committed:
Merge commit '9e6c8953ae16ec088210ef6f2bc68b8a3bd6b786' into A-CheckStyle

XXJJXJ committed:
Merge branch 'master' into A-CheckStyle

XXJJXJ committed:
Merge branch 'master' of https://github.com/XXJJXJ/ip

XXJJXJ committed:
Merge branch 'master' into A-CheckStyle

XXJJXJ committed:
Remove ExitCommand class

XXJJXJ committed:
Update testcases

XXJJXJ committed:
Duke.java: Update methods to account for exit

XXJJXJ committed:
MainWindow.java: Update class fields

XXJJXJ committed:
Ui.java: Remove redundant method

nataniayp commented on src/main/java/duke/exceptions/TaskIsCompleteException.java:
```diff
@@ -0,0 +1,30 @@
+ * This is a TaskIsCompleteException that extends DukeException.
+ * This exception is thrown when the user attempts to mark an already complete task as done.
+ */
+public class TaskIsCompleteException extends DukeException {
```
hello really like how specific the exceptions are! but one thing is that you might want to change it to a noun form (i.e., CompletedTaskException) instead? 😄

XXJJXJ replied:
Thank you for the nice comments! I agree with everything you pointed out. Will work on it!

nataniayp commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
a neat overriding of the method!

nataniayp commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     * @param tasks  An ArrayList of task that contains the keyword in its description.
+     * @return A String of message printed out in CLI.
+     */
+    public String printFindTask(ArrayList<Task> tasks) {
```
no issue with the capitalisation but might want to standardise the method names to `show...` since your methods technically don't print the string 😄

nataniayp commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+    public Deadline(String description, String by) throws CommandParamException {
+        super(description);
+        String[] dateTime = by.trim().split(" ");
+        //index 0 is date, 1 is time
```
you could leave a space between `//` and the comment, but otherwise a good use of comments 👍

nataniayp reviewed (commented):
Overall really good! I'm inspired to have my code as detailed as yours is.

XXJJXJ committed:
Add wrapText field in DialogBox.fxml

XXJJXJ committed:
Rename TaskIsCompleteException to CompletedTaskException

TaskIsCompleteException do not sound appropriate for the exception it is
representing.

It would be more appropriate to have a proper noun name for this
exception case.

TaskIsCompletedException is then renamed to CompletedTaskException.

This new name fully represent the exception with better clarity.

XXJJXJ committed:
Rename methods in Ui.java

New methods created for GUI have different naming convention that starts
with "show" and old methods using CLI starts with "print".

The inconsistencies potentially reduces readability and ease of
understanding for future managers of the code. Hence a standardization
is required.

Rename all "print" prefix methods to "show" prefix with minor
adjustment to some method names.

Using "show" is more suitable to describe the actions of duke because
Duke now runs on GUI instead of CLI.

XXJJXJ committed:
Update comments in Deadline and Event class

XXJJXJ committed:
Refactor Duke and MainWindow class

XXJJXJ committed:
Add assertions for ToDo class

XXJJXJ committed:
Add assertions for Event class

XXJJXJ committed:
Add assertions for Deadline class

XXJJXJ committed:
Improve readability of AddCommand class

XXJJXJ committed:
Add assertions to Storage class

XXJJXJ committed:
Improve readability of Storage class

XXJJXJ committed:
Add assertions to TaskList class

XXJJXJ committed:
Update Javadocs in TaskList class

XXJJXJ committed:
Update Javadocs in Ui class

XXJJXJ committed:
Update Javadocs for Duke class

XXJJXJ committed:
Fix bugs with assertion in Storage class

XXJJXJ committed:
Merge pull request #1 from XXJJXJ/branch-A-Assertions

[Branch-A-assertions]

XXJJXJ committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/util/Storage.java

XXJJXJ committed:
Implement more error handling for done and delete commands

XXJJXJ committed:
Fix CheckStyle violation in ParserTest

XXJJXJ committed:
Create gradle.yml

XXJJXJ committed:
Modify Task to implement Comparable

XXJJXJ committed:
Add abstract class TaskWithDateTime

XXJJXJ committed:
Refactor Deadline and Event class to extend TaskWithDateTime

XXJJXJ committed:
Add compareTo method in ToDo

XXJJXJ committed:
Add SortCommand.java

XXJJXJ committed:
Add Collections import in TaskList for sorting

XXJJXJ committed:
Fix bugs with FindCommand

XXJJXJ committed:
Merge pull request #2 from XXJJXJ/branch-A-CI

Create gradle.yml

XXJJXJ committed:
Add more cases into Parser class

XXJJXJ committed:
Recover Cli functionality in Duke

XXJJXJ committed:
Add scanner into Ui class for Cli users

XXJJXJ committed:
Recover ExitCommand class for Cli users

XXJJXJ committed:
Merge branch 'branch-C-FriendlierSyntax'

XXJJXJ committed:
Add alternative syntax for bye from GUI

XXJJXJ committed:
Merge branch 'master' into branch-C-Sort

XXJJXJ committed:
Fix CheckStyle violation in TaskWithDateTime class

XXJJXJ committed:
Merge pull request #3 from XXJJXJ/branch-C-Sort

Branch-C-Sort

XXJJXJ committed:
Refactor TaskWithDateTime method parameters

XXJJXJ committed:
Add Javadocs to SortCommand and TaskWithDateTime

XXJJXJ committed:
Fix compareTo method of ToDo and TaskWithDateTime

XXJJXJ committed:
Split DialogBox into more classes

XXJJXJ committed:
Increase size of MainWindow

XXJJXJ committed:
Fix code style in MainWindow.fxml

XXJJXJ committed:
Add title to GUI window

XXJJXJ committed:
Add Ui.png

XXJJXJ committed:
Update README.md

XXJJXJ committed:
Modify style in DukeExceptionDialogBox.fxml

XXJJXJ committed:
Update Ui.png

XXJJXJ committed:
Update README.md

XXJJXJ committed:
Set theme jekyll-theme-architect

XXJJXJ committed:
Set theme jekyll-theme-hacker

XXJJXJ committed:
Set theme jekyll-theme-hacker

XXJJXJ committed:
Remove Ui.png

XXJJXJ committed:
Merge branch 'master' of https://github.com/XXJJXJ/ip

XXJJXJ committed:
Add Ui.png

XXJJXJ committed:
Remove CLI feature

XXJJXJ committed:
Fix magic literals in Duke.java

XXJJXJ committed:
Add GUI window icon

XXJJXJ committed:
Modify avatar png files to 1:1 ratio

XXJJXJ committed:
Add circle to profile picture

XXJJXJ committed:
Update DukeExceptionDialog.fxml

XXJJXJ committed:
Update Ui.png

XXJJXJ committed:
Update GUI

XXJJXJ committed:
Update Ui

XXJJXJ committed:
Update README.md

XXJJXJ committed:
Update docs/README.md

XXJJXJ committed:
Update docs/README.md

XXJJXJ committed:
Merge pull request #5 from XXJJXJ/branch-TOC-tags

Branch toc tags
